### PR TITLE
Closes #1330 : Clone the Categories explorer into an Additive Explorer 

### LINKED ISSUE #1330: Clone the Categories explorer into an Additive Explorer
Duplicating the Categories explorer to let users explore additives.

### PR DESCRIPTION
**Pull request by PrajwalM2212**
## Description

Users can now explore additives

## Related issues and discussion
#1330 
 
 ## Screen-shots, if any
 
 
add

screenshot_1523902166
screenshot_1523902171

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
@teolemon Created the additives explorer

**Comment by teolemon** [comment]
Can you sort them by E-number (from 100 to 1000)

**Comment by PrajwalM2212** [comment]
@teolemon 
image
Is this okay ? I have sorted them based on the numbers following E .

**Comment by PrajwalM2212** [comment]
screenshot_1523972904
@teolemon Is this icon okay?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        progressBar = findViewById(R.id.additives_progress_bar);
+        fastScroller = findViewById(R.id.additives_fast_scroller);
+
+
+        progressBar.setVisibility(View.VISIBLE);
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url("https://world.openfoodfacts.org/additives.json")
+                .build();
+
```
Comment (jaztriumph) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java]: @PrajwalM2212 there is no need for downloading data everytime as the additivies are already stored in local database in `AdditiveNmaeDao` and you can extract data from that using the `getAdditiveByTagAndLanguageCode` function in `ProductRepository.java` this even takes care of translations. By this you code will just become very compact.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Additives.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package openfoodfacts.github.scrachx.openfood.models;
```
Comment (jaztriumph) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Additives.java]: @PrajwalM2212 {read the below comment first} as the `AdditiveNmaeDao` returns `AdditiveName` there is no need for this class, it is just redundant.

**Comment by PrajwalM2212** [comment]
@jaztriumph Thanks ☺️. That was really helpful and also made the code compact.

**Comment by teolemon** [comment]
What does it look if the EXXX is in white, like the other icons ? Is that legible ?

**Comment by PrajwalM2212** [comment]
@teolemon 
image

**Comment by PrajwalM2212** [comment]
I have added the filter functionality

**Comment by teolemon** [comment]
@PrajwalM2212 conflict to fix there.

**Comment by PrajwalM2212** [comment]
The project has undergone some major changes . RxJava and MVVM has been used extensively.
It will take some time to update this PR .

**Comment by Karljoones** [comment]
Hi @PrajwalM2212 I am not sure if you were one of the students who mentioned they were doing exams at the moment - if you were not, is there any update on this PR?

**Comment by PrajwalM2212** [comment]
@Karljoones No it was not me . I will update the PR when i am free.

**Comment by PrajwalM2212** [comment]
Will reopen the PR after I make changes

**Comment by Karljoones** [comment]
@PrajwalM2212 no worries!

**Comment by teolemon** [comment]
@PrajwalM2212 Can you reopen the Pull Request ?

**Pull request closed without merging**